### PR TITLE
chore(operator): Update build and runtime deps

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Regular upgrade cycle before releasing the next Community Loki Operator version:

| Dependency | Old version | New Version | Requires Changes |
|--------|--------|--------|--------|
| go | `v1.21` | `v1.22.8` |:heavy_check_mark: |
| k8s | `1.28.7`  | `v1.30.5` |:heavy_check_mark: |
| controller-runtime | `v0.16.5` | `v0.18.5` ||
| Prometheus APIs | `0.71.2` | `v0.76.1` | :heavy_check_mark:  |
| Prometheus Client Golang | `v1.19.0` | `v1.20.4` ||
| Operator SDK| `v1.33.0` | `v1.33.0`||
| GolangCI-lint | `v1.56.2` | `v1.61.0`||
| Gofumpt | `v0.6.0` | `v0.7.0`||
| Kind | `v0.22.0` | `v0.22.0`||

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:
- No update of operator-sdk because it requires heavy changes to adopt the go/v4 kubebuilder project layout.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Review by JoaoBraveCoding — APPROVED** [review]
lgtm

File: operator/config/crd/bases/loki.grafana.com_lokistacks.yaml [hunk 1]
```diff
@@ -3891,6 +3890,7 @@ spec:
                 - mode
                 type: object
             required:
+            - managementState
```
Comment (xperimental) [inline operator/config/crd/bases/loki.grafana.com_lokistacks.yaml:3893]: I think we want to change `managementState` to be `optional` like it was always treated in the previous versions. There is a default for it, so the API _should_ fill in the value automatically.

File: operator/config/crd/bases/loki.grafana.com_lokistacks.yaml [hunk 2]
```diff
@@ -5084,6 +5070,7 @@ spec:
...
```
Comment (xperimental) [inline operator/config/crd/bases/loki.grafana.com_lokistacks.yaml:5073]: I think we want to change `managementState` to be `optional` like it was always treated in the previous versions. There is a default for it, so the API _should_ fill in the value automatically.

File: operator/config/crd/bases/loki.grafana.com_lokistacks.yaml [hunk 3]
```diff
@@ -224,14 +223,15 @@ spec:
                                       description: Regex allows choosing the attributes
                                         by matching a regular expression.
                                       type: string
+                                  required:
+                                  - action
```
Comment (xperimental) [inline operator/config/crd/bases/loki.grafana.com_lokistacks.yaml:227]: This change is correct, we just forgot to remove the `omitempty` from one of the `action` definitions. This is part of the OTLP attribute configuration, so this _maybe_ changes soon anyway.

File: operator/internal/certrotation/rotation.go [hunk 4]
```diff
@@ -74,7 +74,7 @@ func (r *certificateRotation) NewCertificate(signer *crypto.CA, validity time.Du
 		return nil
 	}
 
-	return signer.MakeServerCertForDuration(sets.NewString(sets.List[string](r.Hostnames)...), validity, addClientAuthUsage, addSubject)
+	return signer.MakeServerCertForDuration(sets.Set[string](r.Hostnames), validity, addClientAuthUsage, addSubject)
```
Comment (xperimental) [inline operator/internal/certrotation/rotation.go:77]: I'm getting `redundant type conversion` on this line:

```suggestion
	return signer.MakeServerCertForDuration(r.Hostnames, validity, addClientAuthUsage, addSubject)
```

**Review by xperimental — APPROVED** [review]
Seems to work fine per my test, only have two small comments/questions...

**Pull request merged**